### PR TITLE
[0.25->0.26] Port telemetry changes for tracking clientIds from previous connections

### PR DESCRIPTION
**Pull request by heliocliu**
Bringing changes originally made in #3613 to 0.26 branch for patch release.
#3605

### CODE REVIEW

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -1385,6 +1397,12 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         if (value === ConnectionState.Connected) {
             this._clientId = this.pendingClientId;
+            if (this._clientId !== undefined) {
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:1400]: It is never undefined here.

Comment (heliocliu) [reply packages/loader/container-loader/src/container.ts:1400]: The linter got mad at me for using the non-null assertion.  Should have just added the ignore I guess 🙃

**Comment by vladsud** [comment]
I think this kind of logging can be done more elegantly.
I'd not track 100 clients - we can avoid it via following 2 checks:

1. Any incoming message should be from clientID that is in quorum.
2. With having container session ID added to IClient (PR that Tony opened), we can assert (when receiving any join op) that there is exactly one clientId for any given containerID in quorum.

Last point above is a bit nuanced, as client may believe that it lost connection and thus it creates new connection, while old one is still there (from server POV) and will eventually time-out and get closed. We can work around it by
2. marking our own old clientID "dead" in quorum on "connected" event and asserting that ops are never sent for dead client. Note that with this modification, we do not require container session ID concept.

The reason I prefer this approach over one in PR and main is that we do not need to allocate a bunch of memory, and we also do not need to scan that much memory on any op. In worst case we are talking about 16Kb scan on every op - this is not cheap.
While quorum can have a lot of members, it is a hash map (under the cover), so checking client is relatively cheap, and does not use more memory.

**Comment by heliocliu** [comment]
@vladsud I think the proposed checks would be good to get into main after Tony's change, but these two changes are primarily for collecting telemetry first.  We also don't know the scale/exact cause of the issue so we wanted to avoid changing the behavior too much.  I can reduce the number of tracked clients though to like 20?

**Comment by vladsud** [comment]
We do not need Tony's change to IClient - see my clarification with marking clients as "dead" in quorum.
I've described it in more concise form in https://github.com/microsoft/FluidFramework/issues/3675

Let's move with this change, but I'd prefer we rework it in main (let's start with discussion in above issue)

**Review by vladsud — APPROVED** [review]

**Review by vladsud — APPROVED** [review]

**Pull request merged**